tests/render/util.py

```diff
@@ -0,0 +1,130 @@
+import copy
+from typing import Dict, List, Optional
+
+import nbformat
+from nbconvert.preprocessors import ExecutePreprocessor
+from nbformat.notebooknode import NotebookNode
```

Comment:
These were mixed in with actual tests for utils. I moved to its own file, and adjusted imports accordingly